File: DSA/RECURSION/multiply.c
#include<stdio.h>
int multiply(int n,int m)
{
    if(n==0 || m==0)
    {
        return 0;
    }
    else if(m==1)
    {
        return n;
    }
    else
    {
        return (multiply(n,m-1)+n);
    }
}
void main()
{
    int n,m;
    printf("Enter two numbers\n");
    scanf("%d%d",&n,&m);
    printf("product of these two numbers is %d\n",multiply(n,m));
}

File: Java/JAVA 3.0/RECURSION/multiply.java
package RECURSION;

public class multiply {
    static int multiply(int x,int y)
    {
        if(y==1) return x;
        return multiply(x,y-1) + x;
    }

    public static void main(String[] args) {
        System.out.println(multiply(5,3));
    }
}
